tool.go

```diff
@@ -0,0 +1,269 @@
+		logrus.Fatal("Error writing the output file, quit")
+	}
+	defer output.Close()
+
+	data := map[string]interface{}{
+		"serviceConstructs": serviceConstructs,
+		"usesNetworks":      usesNetworks,
+		"usesVolumes":       usesVolumes,
+	}
+	outputTemplate := template.Must(template.New("output_template").Parse(outputTemplate))
+	if err := outputTemplate.Execute(output, data); err != nil {
+		logrus.Fatal("Error writing the output template to file: %v", err)
+	}
+
+	logrus.Infof("---Please check the output.txt file for list of Docker/Rancher 1.6 constructs needing special handling---")
+}
+
+func lookForDocumentedConstructs(reflectedConfig reflect.Value, inputConstructsMap map[string]DocConstruct, constructsOut map[string][]DocConstruct) {
+	for key, doc := range inputConstructsMap {
+		f := reflectedConfig.FieldByName(key)
+		if f.IsValid() && !isEmpty(f) {
+			if doc.Special {
+				outDoc := handleSpecialConstruct(key, f)
+				constructsOut[doc.Name] = append(constructsOut[doc.Name], outDoc...)
+			} else {
+				constructsOut[doc.Name] = append(constructsOut[doc.Name], doc)
+			}
+		}
+	}
+}
+
+func isEmpty(reflectedField reflect.Value) bool {
+	fieldType := reflectedField.Type()
+	switch fieldType.Name() {
+	case "string":
+		return reflectedField.Interface().(string) == ""
+	default:
+		return reflectedField.IsNil()
+	}
+
+}
+
+func handleSpecialConstruct(key string, reflectedField reflect.Value) []DocConstruct {
+
+	var outDoc []DocConstruct
+
+	switch key {
+	case "Labels":
+		labels, ok := reflectedField.Interface().(map[string]string)
+		if !ok {
+			logrus.Infof("\n--- Error reading labels from compose config file")
+			return outDoc
+		}
+		if labels != nil && len(labels) != 0 {
+			for labelKey, _ := range labels {
+				if strings.HasPrefix(labelKey, "io.rancher.scheduler") {
+					doc := DocConstruct{
+						Name:     labelKey,
+						DocLink:  "https://rancher.com/blog/2018/2018-08-29-scheduling-options-in-2-dot-0/",
+						HelpText: "Please refer to the scheduling options in Rancher 2.0 to migrate this functionality",
+					}
+					outDoc = append(outDoc, doc)
+				} else {
+					doc := DocConstruct{
+						Name: labelKey,
+					}
+					outDoc = append(outDoc, doc)
+				}
+			}
+		}
+	}
+
+	return outDoc
+}
+
+func readDockerCompose(t *ToolArgs) *ComposeConfig {
+	dcBytes, err := ioutil.ReadFile(t.dockerComposeFile)
+	if err != nil {
+		logrus.Infof("Failed to read file %v, error: %v ", t.dockerComposeFile, err)
+		return nil
+	}
+
+	dockerConfig := &ComposeConfig{}
+
+	err = yaml.Unmarshal(dcBytes, &dockerConfig)
+	if err != nil {
+		logrus.Infof("Failed to parse the yaml file, error: %v", err)
+		return nil
+	}
+	return dockerConfig
+}
+
+func readRancherCompose(t *ToolArgs) *RancherComposeConfig {
+	if t.rancherComposeFile == "" {
+		return nil
+	}
+	dcBytes, err := ioutil.ReadFile(t.rancherComposeFile)
+	if err != nil {
+		logrus.Infof("Failed to read file %v, error: %v ", t.rancherComposeFile, err)
+		return nil
```

Comment:
it also feels like we should return any kinds of errors to the caller stack, and make a decision there based on it. Most likely an error would be an indication for the tool to exit